### ISSUE TITLE
Capturing user in Flask

### ISSUE THREAD

**Issue by kam193** [getsentry/sentry-python#769]
tl;dr: What is the recommended way to capture user in Flask, in case when there is no `flask-login`?

First try is to configure scope, so maybe decorate view with `push_scope`? Nope - it looks nice, but it is not enough if you want to have user info during e.g. error handling, which could be outside view method.

So we have `configure_scope`, what tell us the [Sentry doc](https://docs.sentry.io/platforms/python/#capturing-the-user)? Great, but only before #758 - before this PR, `FlaskIntegration` cleared the scope on every app context change (so typically for every request). Now information set global can be shared during requests - not the best option for user info.

And the last try, we have `before_send` hook, what is currently probably the only way to capture user. Ok, but it's like totally different use as designed for it (stripping sensitive data), and this way is undocumented neither in general Sentry doc, nor in Flask-specific page (as an option for capturing user).

I think we should have a recommended way for capturing user in Flask: by scope (similar like other platforms), hook (which works now) or a special one (like new hook in FlaskIntegration instead of hard coded support for `flask-login` only).

**Comment by untitaker** [comment]
You can use `configure_scope()` inside of `@app.before_request`, your view, or a wsgi middleware. If the scope is set during request handling, the data is not shared across requests.

```python
@app.before_request
def _before(*a, **kw):
    with sentry_sdk.configure_scope() as scope:
         scope.user = {"id": request.environ['????']}
```

Hope that helps

**Comment by kam193** [comment]
Oh, ok, it looks like I misunderstand the role of #758, the global scope is still limited to the current context by `_request_started` - I missed this.

So, issue to close ;) But can I document this behavior in https://docs.sentry.io/platforms/python/flask/? I think it's important information

**Comment by untitaker** [comment]
Does this work? https://github.com/getsentry/sentry-docs/pull/1937

**Comment by kam193** [comment]
Yeah, this is what I meant :)
